aten/src/ATen/core/op_registration/op_registration_test.cpp

```diff
@@ -906,6 +972,65 @@ TEST(OperatorRegistrationTest, AutogradBackendOverridesAutogradKernel) {
   EXPECT_FALSE(called_nonautograd);
 }
 
+TEST(OperatorRegistrationTest, AutogradBackendOverridesMathKernel) {
```

Comment:
If I'm right about the need to avoid choosing Autograd when no backend kernel is registered, we should probably add (or replace this with) a test for that.